pkg/services/sqlstore/alert.go

```diff
@@ -251,11 +259,15 @@ func deleteMissingAlerts(alerts []*m.Alert, cmd *m.SaveAlertsCommand, sess *DBSe
 		}
 
 		if missing {
-			deleteAlertByIdInternal(missingAlert.Id, "Removed from dashboard", sess)
+			if err := deleteAlertByIdInternal(missingAlert.Id, "Removed from dashboard", sess); err != nil {
+				if gotErr == nil {
```

Comment:
Same as above: is this condition necessary? Is `gotErr` going to be something other than nil?

Reply:
@papagian Pretty sure it is, see above answer please.

Reply:
👍 I see, since it's inside a loop I think it is better to return directly the error instead of waiting for the loop to finish.